.clang-tidy

```diff
@@ -0,0 +1,12 @@
+Checks:
+  *,
+  -altera*,
+  -llvm*,
```

Comment:
My understanding is this disables the LLVM coding standards. Wouldn't we want something like that?

Or are we just using Google's standard. In that case, why don't we just targettedly enable Google's? Or have we not decided?

Reply:
Some clang-tidy checks are super opinionated and are really for internal use by various orgs. E.g. iirc altera does lots of low-level embedded stuff where they hate dynamic loop bounds and always want loops to be unrolled if possible.

Reply:
Why are we blanket-enabling everything and then disabling select things?

I feel like we should do the opposite. Enable only what we want.

Reply:
Added comments explaining.